### ISSUE TITLE
Instance ID (AWS EC2) in security_group_ingress_rules.sql

### ISSUE THREAD

**Issue by rohinihirurkar** [cloudquery/cloudquery#1736]
### Why do you need this resource or plugin? Please describe.
Currently I am able to pull the details using security_group_ingress_rules.sql but since there are lot of resources as part of the result  its difficult to find which ec2 instance the port is open associated to the listed security group
### Describe the solution you'd like
So from this security_group_ingress_rules.sql is there possibilities to fetch ec2 instance ID/details as well . For example how to see port 22 is open for which ec2 instance 

Thanks in advance

**Comment by bbernays** [comment]
You can join the ARN column in that view with the security groups attached to an EC2 Instance like this:

``` sql
SELECT AWS_EC2_INSTANCES.ARN AS EC2_ARN,
	VIEW_AWS_SECURITY_GROUP_INGRESS_RULES.*
FROM AWS_EC2_INSTANCES
LEFT JOIN AWS_EC2_INSTANCE_NETWORK_INTERFACES ON AWS_EC2_INSTANCES.CQ_ID = AWS_EC2_INSTANCE_NETWORK_INTERFACES.INSTANCE_CQ_ID
LEFT JOIN AWS_EC2_INSTANCE_NETWORK_INTERFACE_GROUPS ON AWS_EC2_INSTANCE_NETWORK_INTERFACE_GROUPS.INSTANCE_NETWORK_INTERFACE_CQ_ID = AWS_EC2_INSTANCE_NETWORK_INTERFACES.CQ_ID
LEFT JOIN VIEW_AWS_SECURITY_GROUP_INGRESS_RULES ON VIEW_AWS_SECURITY_GROUP_INGRESS_RULES.ARN = 'arn:aws:ec2:' || AWS_EC2_INSTANCES.REGION || ':' || AWS_EC2_INSTANCES.ACCOUNT_ID || ':security-group/' || AWS_EC2_INSTANCE_NETWORK_INTERFACE_GROUPS.GROUP_ID
```

The only caveat is that this will only work in the AWS commercial regions (not AWS Gov Cloud or the China regions)

**Comment by rohinihirurkar** [comment]
Thank you for the query provided and compared with with results from above query vs security_group_ingress_rules.sql some of the information are not resulting. Here is what i was looking for

List of all security groups and rules along with ec2 instances , other resources , 
for example I want to see port 22 open  (ingress rules) which should give me all the columns in security_group_ingress_rules.sql along with ec2 instances ID, other (LoadBalancer,EMR Cluseter,RDS etc..)resources ID's
Thanks in advance

**Comment by cqgaurav** [comment]
Hi Rohini, thanks for your report. Could you let us know if some columns are missing when you sync, or if this is a feature request for more columns. I am closing this for now, but please feel free to reopen with more details.

**Comment by rohinihirurkar** [comment]
> 

Hi cqgaurav, 

I have reinstalled the CQ with new version and the above query is not working , Here is the all ec2 instance associated with ingress rules along with region

create view ec2instance__aws_security_group_ingress_rules as(
SELECT VIEW_AWS_SECURITY_GROUP_INGRESS_RULES.*, AWS_EC2_INSTANCES.ARN AS EC2_ARN, AWS_EC2_INSTANCES.ID as EC2_ID FROM VIEW_AWS_SECURITY_GROUP_INGRESS_RULES
LEFT JOIN aws_ec2_instance_security_groups ON VIEW_AWS_SECURITY_GROUP_INGRESS_RULES.id = aws_ec2_instance_security_groups.group_id
LEFT JOIN aws_ec2_instances on (aws_ec2_instance_security_groups.instance_cq_id = AWS_EC2_INSTANCES.CQ_ID and AWS_EC2_INSTANCES.REGION = VIEW_AWS_SECURITY_GROUP_INGRESS_RULES.region)
);

But above query is not working as I see the new CQ has changed with DB structure

will you be able to provide same with new DB query

**Comment by erezrokah** [comment]
Going to re-open per https://github.com/cloudquery/cloudquery/issues/1736#issuecomment-1294546119 so we can investigate

**Comment by bbernays** [comment]
This issue (columns/tables changing from V0 to V1) will be caught and fixed when we complete #2190

**Comment by jsonpr** [comment]
Query was provided in Discord here: https://discord.com/channels/872925471417962546/1016576453988601946

`SELECT AWS_EC2_SECURITY_GROUPS.*, AWS_EC2_INSTANCES.instance_id, AWS_EC2_INSTANCES.arn as instance_arn FROM AWS_EC2_INSTANCES
CROSS JOIN jsonb_array_elements(security_groups) ec2_json
RIGHT JOIN AWS_EC2_SECURITY_GROUPS on ec2_json.value ->> 'GroupId' = AWS_EC2_SECURITY_GROUPS.group_id;`

This will provide a list of Security Groups with relevant EC2 instances (Instance IDs and Instance ARNs).  The Security Group ingress rules will be in the ip_permissions column.  For AWS_EC2_SECURITY_GROUPS reference: https://github.com/cloudquery/cloudquery/blob/main/plugins/source/aws/docs/tables/aws_ec2_security_groups.md
